## Importing Libraries

In [1]:
import pandas as pd
import numpy as np

## Reading the Dataset

In [2]:
# Reading X_train dataset
data1 = pd.read_csv(r'X_train.csv', header='infer')

In [3]:
# Reading the y_train dataset
data2 = pd.read_csv(r'y_train.csv', header='infer')

In [5]:
# Merging the datasets
data = data1.merge(data2, on=['playerId'], how='inner')
print(data.head())

   playerId           Name Sex Equipment   Age  BodyweightKg BestSquatKg  \
0   19391.0   Carlos Ceron   M       Raw  23.0         87.30       205.0   
1   15978.0   Tito Herrera   M     Wraps  23.0         73.48       220.0   
2   27209.0    Levi Lehman   M       Raw  26.0        112.40       142.5   
3   27496.0  Stacy Hayford   F       Raw  35.0         59.42        95.0   
4   20293.0  Brittany Hirt   F       Raw  26.5         61.40       105.0   

   BestDeadliftKg  BestBenchKg  
0           235.0        125.0  
1           260.0        157.5  
2           220.0        145.0  
3           102.5         60.0  
4           127.5         60.0  


In [6]:
data.describe()

playerId          Age  BodyweightKg  BestDeadliftKg   BestBenchKg
count  18900.00000  18725.00000  18900.000000     18900.00000  18900.000000
mean   15039.49963     29.66470     85.425557       201.12277    116.963389
std     8674.67268     11.55708     22.959720        62.17163     51.231651
min        0.00000      7.00000     26.130000        18.10000      9.100000
25%     7462.75000     21.50000     67.700000       149.85750     72.500000
50%    15122.50000     26.50000     82.100000       204.12000    115.000000
75%    22540.25000     35.00000     98.970000       247.50000    150.000000
max    29998.00000     83.00000    201.000000       408.23000    425.000000

## Handling Null Values

In [7]:
data.isnull().sum()

playerId            0
Name                0
Sex                 0
Equipment           0
Age               175
BodyweightKg        0
BestSquatKg         0
BestDeadliftKg      0
BestBenchKg         0
dtype: int64

We can see that the Age column has 175 missing values. We will fill in those missing values,
but to determine how to fill in the column, we need to know the data type of that column.
To check the data type of the attributes. .info() function is used.

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18900 entries, 0 to 18899
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   playerId        18900 non-null  float64
 1   Name            18900 non-null  object 
 2   Sex             18900 non-null  object 
 3   Equipment       18900 non-null  object 
 4   Age             18725 non-null  float64
 5   BodyweightKg    18900 non-null  float64
 6   BestSquatKg     18900 non-null  object 
 7   BestDeadliftKg  18900 non-null  float64
 8   BestBenchKg     18900 non-null  float64
dtypes: float64(5), object(4)
memory usage: 1.3+ MB


In [9]:
data['Age'].fillna(data['Age'].mean(),inplace=True)

In [10]:
data.isnull().sum()

playerId          0
Name              0
Sex               0
Equipment         0
Age               0
BodyweightKg      0
BestSquatKg       0
BestDeadliftKg    0
BestBenchKg       0
dtype: int64

## Label Encoding for Sex and Equipment

In [11]:
# converting the sec column object type to float type
data['Sex'] = data['Sex'].map({"M":1, "F":0})
#encode the equipment column
from sklearn.preprocessing import LabelEncoder
data["Equipment"] = LabelEncoder().fit_transform(data['Equipment'])

In [12]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18900 entries, 0 to 18899
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   playerId        18900 non-null  float64
 1   Name            18900 non-null  object 
 2   Sex             18900 non-null  int64  
 3   Equipment       18900 non-null  int32  
 4   Age             18900 non-null  float64
 5   BodyweightKg    18900 non-null  float64
 6   BestSquatKg     18900 non-null  object 
 7   BestDeadliftKg  18900 non-null  float64
 8   BestBenchKg     18900 non-null  float64
dtypes: float64(5), int32(1), int64(1), object(2)
memory usage: 1.2+ MB


In [13]:
data.describe()

playerId           Sex     Equipment           Age  BodyweightKg  \
count  18900.00000  18900.000000  18900.000000  18900.000000  18900.000000   
mean   15039.49963      0.675714      1.524127     29.664700     85.425557   
std     8674.67268      0.468120      0.839712     11.503448     22.959720   
min        0.00000      0.000000      0.000000      7.000000     26.130000   
25%     7462.75000      0.000000      1.000000     21.500000     67.700000   
50%    15122.50000      1.000000      1.000000     26.500000     82.100000   
75%    22540.25000      1.000000      2.000000     34.500000     98.970000   
max    29998.00000      1.000000      3.000000     83.000000    201.000000   

       BestDeadliftKg   BestBenchKg  
count     18900.00000  18900.000000  
mean        201.12277    116.963389  
std          62.17163     51.231651  
min          18.10000      9.100000  
25%         149.85750     72.500000  
50%         204.12000    115.000000  
75%         247.50000    150.000000  
max         408.23000    425.000000

In [15]:
data.shape

(18900, 9)